sandbox/test_print.cpp

```diff
@@ -39,219 +39,8 @@
-         (*it)[4] = 0. ; // TO DO, (*it)[4] ^= (*it)[4] XOR, or basic set up to 0 <--------------------------------- TIM
-         (*it)[3] = 0. ;
-         cnrn_rates(it);
-         (*it)[3] = (*it)[6];
-         (*it)[4] = (*it)[10];
-     }
-
-     template<class iterator, cyme::order O>
-     static inline void cnrn_cur(iterator it){
-         double _v=-33.33; // _v voltage fron a node random number
-         cyme::serial<value_type,O> tmp  = cnrn_current<iterator, O>(it, _v + 0.001);
-         cyme::serial<value_type,O> tmp2 = cnrn_current<iterator, O>(it, _v);
- //        tmp() -= tmp2();
-         (*it)[17] = tmp;
-         (*it)[17] = (*it)[17]/0.001;
-     }
-
-*/
-     template<class iterator>
-     static inline void cnrn_state(iterator it){
-         cnrn_states(it);
-     }
-     const static size_t value_size = 18;
- };
-
-
- template<class mechanism, class container = cyme::vector<mechanism, cyme::AoSoA> > // should use cyme container there
- class pack{
- public:
-     typedef typename mechanism::value_type value_type;
-
-     explicit pack(size_t s, size_t value):cont(s,value){ // vector constructor
-     }
-
-     explicit pack():cont(){ // array constructor
-     }
-
-     void execution(){
-         for(typename container::iterator it = cont.begin(); it < cont.end(); ++it)
-             m.template cnrn_functions<typename container::storage_type, container::order_value>(*it);
-     }
-
-     value_type& operator()(size_t i, size_t j){
-         return cont(i,j);
-     }
-
-     std::size_t size(){
-         return cont.size();
-     }
-
-     std::size_t size_block(){
-         return cont.size_block();
-     }
-
- private:
-     container cont;
-     mechanism m;
- };
-
- class stack{
- public:
-     typedef boost::function<void(void)> base_type; // pointer to function
-     stack(){
-
-     }
-
-     void push_back(base_type const& f){
-         s.push_back(f); // fill up the stack
-     }
-
-     void flush(){
-         for(std::vector<base_type>::iterator it = s.begin(); it < s.end(); ++it)
-             (*it)(); // execute cnrn_functions
-     }
- private:
-     std::vector<base_type> s;
- };
-
-
- int main(){
-     stack s;
-
-     pack<Na,cyme::vector<Na, cyme::AoSoA> > a(256,0); // pack 16384 synapse, AoSoA
-     pack<Na,cyme::vector<Na, cyme::AoS> > b(256,0); // pack 16384 synapse, AoSoA
-
-     init(a,b);
-
-     s.push_back(boost::bind(&pack<Na,cyme::vector<Na, cyme::AoSoA> >::execution,&a)); // fill up the stack
-     s.push_back(boost::bind(&pack<Na,cyme::vector<Na, cyme::AoS> >::execution,&b)); // fill up the stack
-
-     boost::chrono::system_clock::time_point start =  boost::chrono::system_clock::now();
-     s.flush(); // execute the stack
-     boost::chrono::duration<double>  sec = boost::chrono::system_clock::now() - start;
-     std::cout << " sec " << sec.count() << std::endl;
-
-     check(a,b);
- }
+int main(){
+	cyme::vec_simd <float,cyme::vmx,4> log2e(1.4426950408889634073599);
```

Comment:
do not hard vmx, it creates bug if you compile this on Power use instead __GETSIMD__()  it will the correct architecture